naga/src/back/spv/writer.rs

```diff
@@ -2710,6 +2777,41 @@ impl Writer {
     pub(super) fn needs_f16_polyfill(&self, ty_inner: &crate::TypeInner) -> bool {
         self.io_f16_polyfills.needs_polyfill(ty_inner)
     }
+
+    #[allow(dead_code)]
```

Comment:
When would this be dead? Lets try to use a conditional expect to encode when this should happen.